lib/rubocop/cop/style/redundant_parentheses.rb

```diff
@@ -29,6 +29,9 @@ class RedundantParentheses < Base
         # @!method rescue?(node)
         def_node_matcher :rescue?, '{^resbody ^^resbody}'
 
+        # @!method pin_operator?(node)
+        def_node_matcher :pin_operator?, '^(pin (begin !{lvar ivar cvar gvar}))'
```

Comment:
The matcher name `pin_operator?` is abstract. How about the names `allowed_pin_operator?` or `pin_operator_with_var`?